packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -63,18 +63,19 @@ const App = (props) => {
     const location = useLocation()
     const authModal = useAuthModal()
     const customer = useCustomer()
+    const appConfig = useAppConfig()
 
-    const site = useSite()
-    const locale = useLocale()
+    const {pathname, search} = useLocation()
+    const site = useMemo(() => {
+        return resolveSiteFromUrl(`${pathname}${search}`)
+    }, [pathname, search])
+    const locale = useMemo(() => {
+        return resolveLocaleFromUrl(`${pathname}${search}`)
```

Comment:
These functions already use useMemo to memoize the return value. Is it necessary to use another useMemo for them here?